<!DOCTYPE html>
<html lang="en">
<head>
  
  
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    
    
    
    <link rel="shortcut icon" href="../../img/favicon.ico">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
    <title>CNV Pytor - igv.js</title>
    <link href="../../css/bootstrap-3.3.7.min.css" rel="stylesheet">
    <link href="../../css/font-awesome-4.7.0.css" rel="stylesheet">
    <link href="../../css/base.css" rel="stylesheet">
    <link rel="stylesheet" href="../../css/highlight.css">
    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
        <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
        <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <script src="../../js/jquery-3.2.1.min.js"></script>
    <script src="../../js/bootstrap-3.3.7.min.js"></script>
    <script src="../../js/highlight.pack.js"></script>
    
    <base target="_top">
    <script>
      var base_url = '../..';
      var is_top_frame = false;
        
        var pageToc = [
          {title: "File formats", url: "#_top", children: [
              {title: "pytor", url: "#pytor" },
              {title: "vcf", url: "#vcf" },
          ]},
          {title: "Configuration Options", url: "#configuration-options", children: [
              {title: "Examples", url: "#examples" },
          ]},
        ];

    </script>
    <script src="../../js/base.js"></script> 
</head>

<body>
<script>
if (is_top_frame) { $('body').addClass('wm-top-page'); }
</script>



<div class="container-fluid wm-page-content">
  <a name="_top"></a>
    

    <p class="page-title">CNVpytor Track</p>

<p>The CNVpytor track (type = <code>'cnvpytor'</code>) is a tool for copy number analysis from read depth and B-allele frequency 
(BAF) of variants. As input, it takes BAM and/or VCF files. The analyzed data is stored in pytor file and can be viewed 
using <code>cnvpytor</code> track in igv.  One can also use a whole genome VCF file as an input to the <code>cnvpytor</code> track. In such a 
case  read depth and BAF information are calculated on-fly.</p>
<p>For details: https://github.com/abyzovlab/CNVpytor</p>
<h2 id="file-formats">File formats<a class="headerlink" href="#file-formats" title="Direct link to this section">#</a></h2>
<p><code>cnvpytor</code> track can be loaded from either <code>pytor</code> or VCF files.</p>
<h3 id="pytor">pytor<a class="headerlink" href="#pytor" title="Direct link to this section">#</a></h3>
<ul>
<li><a href="https://github.com/abyzovlab/CNVpytor">CNVpytor</a>-produced pytor file.</li>
<li>Option to switch among available bin sizes.</li>
<li>Option to switch among available CNV callers.</li>
<li>Signals: <ul>
<li>Raw Read Depth</li>
<li>GC Corrected Read depth</li>
<li>CNV call </li>
<li>Maximum of BAF likelihood</li>
</ul>
</li>
</ul>
<h3 id="vcf">vcf<a class="headerlink" href="#vcf" title="Direct link to this section">#</a></h3>
<p>Whole genome <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf">VCF</a> file with constraints.</p>
<ul>
<li>Indexed VCF files are not supported as data for the entire genome is required for the calculations. </li>
<li>Both  AD (Read Depth information) and GT (Genotype information) fields are required . </li>
<li>Currently only a single genotype field is supported. Multi-sample VCFs are not supported. </li>
<li>Currently, only ReadDepth CNV caller is available. </li>
<li>Signals:<ul>
<li>Raw Read Depth</li>
<li>Partition</li>
<li>CNV call </li>
<li>Maximum of BAF likelihood</li>
</ul>
</li>
</ul>
<p>Loading the data may take time, as it reads the entire VCF file and calculates the signals.</p>
<h2 id="configuration-options">Configuration Options<a class="headerlink" href="#configuration-options" title="Direct link to this section">#</a></h2>
<p><a href="../Tracks/#options-for-all-track-types">General options</a></p>
<table>
<thead>
<tr>
<th>Property</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td>signal_name</td>
<td>Available Signal names <ul><li>rd_snp : Read Depth and BAF Likelihood</li><li>rd : Read depth</li><li> snp : BAF likelihood</li></ul></td>
<td>rd_snp</td>
</tr>
<tr>
<td>cnv_caller</td>
<td>Name of CNV caller <br>Options<ul><li> ReadDepth: Uses Read depth information only</li><li>2D: Uses both Read depth and BAF information</li></ul>  Shows data based on available caller data.</td>
<td>2D</td>
</tr>
<tr>
<td>bin_size</td>
<td>Bin size <ul> <li>pytor file: Bin size should be avialable in the pytor file</li><li>vcf: Bin size should be multiple of 10,000</li></ul></td>
<td>100000</td>
</tr>
<tr>
<td>colors</td>
<td>Color of the signals. Signal details are in file format section.</td>
<td>['gray', 'black', 'green', 'blue']</td>
</tr>
</tbody>
</table>
<h4 id="examples">Examples<a class="headerlink" href="#examples" title="Direct link to this section">#</a></h4>
<h5 id="pytor-file-as-input">pytor file as input<a class="headerlink" href="#pytor-file-as-input" title="Direct link to this section">#</a></h5>
<pre><code class="language-javascript">{
    id: &quot;pytor_track&quot;,
    type: &quot;cnvpytor&quot;,
    name: &quot;HepG2 pytor&quot;,
    url: &quot;https://storage.googleapis.com/cnvpytor_data/HepG2_WGS.pytor&quot;,
}
</code></pre>
<h6 id="view">View<a class="headerlink" href="#view" title="Direct link to this section">#</a></h6>
<p><img alt="" src="https://github.com/arpanda/igv.js/assets/21196859/8a18f9c0-6dc9-4e21-af66-77d2b2de7e29" /></p>
<h5 id="vcf-file-as-input">VCF file as input<a class="headerlink" href="#vcf-file-as-input" title="Direct link to this section">#</a></h5>
<pre><code class="language-javascript">{
    id: &quot;cnvpytor_track_vcf&quot;,
    type: &quot;cnvpytor&quot;,
    name: &quot;HepG2 VCF&quot;,
    url: &quot;https://storage.googleapis.com/cnvpytor_data/HepG2.vcf.gz&quot;,
}
</code></pre>
<h6 id="view_1">View<a class="headerlink" href="#view_1" title="Direct link to this section">#</a></h6>
<p><img alt="" src="https://github.com/arpanda/igv.js/assets/21196859/b90ebd48-eebd-4d98-af9d-888c365d6e4d" /></p>

  <br>
</div>

<footer class="container-fluid wm-page-content">
  <!--
     <p>
       <a href="https://github.com/igvteam/igv.js-docs/edit/master/docs/tracks/CNVPytor.md"><i class="fa fa-github"></i>
Edit on GitHub</a>
      </p>
  -->
  <!-- <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a> using <a href="https://github.com/gristlabs/mkdocs-windmill">Windmill</a> theme by Grist Labs.</p> -->
</footer>

</body>
</html>