<!DOCTYPE html>
<html lang="en">
<head>
  
  
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    
    
    
    <link rel="shortcut icon" href="../../img/favicon.ico">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
    <title>Alignment - igv.js</title>
    <link href="../../css/bootstrap-3.3.7.min.css" rel="stylesheet">
    <link href="../../css/font-awesome-4.7.0.css" rel="stylesheet">
    <link href="../../css/base.css" rel="stylesheet">
    <link rel="stylesheet" href="../../css/highlight.css">
    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
        <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
        <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <script src="../../js/jquery-3.2.1.min.js"></script>
    <script src="../../js/bootstrap-3.3.7.min.js"></script>
    <script src="../../js/highlight.pack.js"></script>
    
    <base target="_top">
    <script>
      var base_url = '../..';
      var is_top_frame = false;
        
        var pageToc = [
          {title: "File formats", url: "#_top", children: [
          ]},
          {title: "Configuration Options", url: "#configuration-options", children: [
              {title: "colorBy options", url: "#colorby-options" },
              {title: "groupBy options", url: "#groupby-options" },
              {title: "displayMode options", url: "#displaymode-options" },
              {title: "Paired-end and mate-pair coloring options.", url: "#paired-end-and-mate-pair-coloring-options" },
              {title: "Sort option", url: "#sort-option" },
              {title: "Filter options", url: "#filter-options" },
          ]},
          {title: "Functions", url: "#functions", children: [
          ]},
          {title: "Example", url: "#example", children: [
          ]},
        ];

    </script>
    <script src="../../js/base.js"></script> 
</head>

<body>
<script>
if (is_top_frame) { $('body').addClass('wm-top-page'); }
</script>



<div class="container-fluid wm-page-content">
  <a name="_top"></a>
    

    <p class="page-title">Alignment Track</p>

<p>The alignment track (<code>type = 'alignment'</code>) is used to display views of read alignments from BAM or CRAM files.</p>
<h2 id="file-formats">File formats<a class="headerlink" href="#file-formats" title="Direct link to this section">#</a></h2>
<ul>
<li><a href="https://samtools.github.io/hts-specs/SAMv1.pdf">BAM</a></li>
<li><a href="https://samtools.github.io/hts-specs/CRAMv3.pdf">CRAM</a></li>
</ul>
<h2 id="configuration-options">Configuration Options<a class="headerlink" href="#configuration-options" title="Direct link to this section">#</a></h2>
<p><a href="../Tracks/#options-for-all-track-types">General options</a></p>
<table>
<thead>
<tr>
<th>Property</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td>showCoverage</td>
<td>Show coverage depth track.</td>
<td>true</td>
</tr>
<tr>
<td>showAlignments</td>
<td>Show individual alignments.</td>
<td>true</td>
</tr>
<tr>
<td>viewAsPairs</td>
<td>If true, paired reads are drawn connected with a line.</td>
<td>false</td>
</tr>
<tr>
<td>pairsSupported</td>
<td>If false, mate information in paired reads is ignored during downsampling and the 'View as Pairs' option is removed from the alignment track menu.</td>
<td>true</td>
</tr>
<tr>
<td>color</td>
<td>Default color of alignment blocks</td>
<td>rgb(170, 170, 170)</td>
</tr>
<tr>
<td>deletionColor</td>
<td>Color of line representing a deletion</td>
<td>black</td>
</tr>
<tr>
<td>skippedColor</td>
<td>Color of line representing a skipped region (e.g. splice junction)</td>
<td>rgb(150, 170, 170)</td>
</tr>
<tr>
<td>insertionColor</td>
<td>Color of marker for insertions</td>
<td>rgb(138, 94, 161)</td>
</tr>
<tr>
<td>negStrandColor</td>
<td>Color of alignment on negative strand.  Applicable if colorBy = "strand" or "firstOfPairStrand"</td>
<td>rgba(150, 150, 230, 0.75)</td>
</tr>
<tr>
<td>posStrandColor</td>
<td>Color of alignment or position strand.  Applicable if colorBy = "strand" if "firstOfPairStrand"</td>
<td>rgba(230, 150, 150, 0.75)</td>
</tr>
<tr>
<td>pairConnectorColor</td>
<td>Color of connector line between read pairs ("view as pairs" mode).</td>
<td>alignment color</td>
</tr>
<tr>
<td>colorBy</td>
<td>Color alignment by property.  See below.</td>
<td>"unexpectedPair".</td>
</tr>
<tr>
<td>groupBy</td>
<td>Group alignments by property.  See below.</td>
<td></td>
</tr>
<tr>
<td>samplingWindowSize</td>
<td>Window (bucket) size for alignment downsampling in base pairs</td>
<td>100</td>
</tr>
<tr>
<td>samplingDepth</td>
<td>Number of alignments to keep per bucket.  WARNING:  Setting this sampling depth to a high value can freeze the browser when viewing areas of deep coverage.</td>
<td>100.</td>
</tr>
<tr>
<td>alignmentRowHeight</td>
<td>Height in pixels of an alignment row when in expanded mode</td>
<td>14</td>
</tr>
<tr>
<td>readgroup</td>
<td>Readgroup ID value (tag 'RG').</td>
<td></td>
</tr>
<tr>
<td>sort</td>
<td>Initial sort option.  See below.</td>
<td></td>
</tr>
<tr>
<td>filter</td>
<td>Alignment filter options.  See below</td>
<td></td>
</tr>
<tr>
<td>showSoftClips</td>
<td>Show soft-clipped regions</td>
<td>false</td>
</tr>
<tr>
<td>showMismatches</td>
<td>Highlight alignment bases which do not match the reference.</td>
<td>true</td>
</tr>
<tr>
<td>showAllBases</td>
<td>Show all bases of the read sequence.</td>
<td>false</td>
</tr>
<tr>
<td>showInsertionText</td>
<td>Show number of bases for insertions inline when zoomed in.</td>
<td>false</td>
</tr>
<tr>
<td>insertionTextColor</td>
<td>Color for insertion count text.</td>
<td>white</td>
</tr>
<tr>
<td>showDeletionText</td>
<td>Show number of bases deleted inline when zoomed in.</td>
<td>false</td>
</tr>
<tr>
<td>deletionTextColor</td>
<td>Color for deletion count text.</td>
<td>black</td>
</tr>
<tr>
<td>displayMode</td>
<td>See below.</td>
<td>"EXPANDED"</td>
</tr>
<tr>
<td>alignmentRowHeight</td>
<td>Height in pixels for each row of alignments in "EXPANDED" or "FULL" display mode.</td>
<td>14</td>
</tr>
<tr>
<td>squishedRowHeight</td>
<td>Height in pixels for each row of alignments in "SQUISHED" display mode.</td>
<td>3</td>
</tr>
<tr>
<td>coverageColor</td>
<td>Color of coverage track</td>
<td>rgb(150, 150, 150)</td>
</tr>
<tr>
<td>coverageTrackHeight</td>
<td>Height in pixels of the coverage track.</td>
<td>3</td>
</tr>
<tr>
<td>autoscale</td>
<td>Autoscale coverage track to maximum value in view</td>
<td>true unless 'max' is set</td>
</tr>
<tr>
<td>autoscaleGroup</td>
<td>Identifier for an autoscale group for the coverage track.  Tracks with the same identifier are autoscaled together.</td>
<td></td>
</tr>
<tr>
<td>min</td>
<td>Sets the minimum value for the coverage track data scale (y-axis).  Usually zero.</td>
<td>0</td>
</tr>
<tr>
<td>max</td>
<td>Sets the maximum value for the coverage track data scale (y-axis). This value is ignored if autoscale = true</td>
<td>No default</td>
</tr>
</tbody>
</table>
<p><br/></p>
<h4 id="colorby-options">colorBy options<a class="headerlink" href="#colorby-options" title="Direct link to this section">#</a></h4>
<ul>
<li>none - default color</li>
<li>strand - Color by read strand</li>
<li>firstOfPairStrand - For paired alignments.  Color both mates in a pair by the read strand of the mate marked "first in pair"</li>
<li>pairOrientation - For paired alignments.  Color by relative orientation (strand) of mates to each other</li>
<li>tlen - For paired alignments.  Color by the "template length (insert size)" of the pair, as indicated by the tlen property of the alignment.  </li>
<li>unexpectedPair - A combination of <code>pairOrientation</code> and <code>tlen</code> options</li>
<li>tag:<em>tagName</em> - Color alignments by the tag indicated,  e.g.  tag:YC.</li>
<li>basemod - Color alignments by base modification.  See the <a href="https://igv.org/doc/desktop/#UserGuide/tracks/alignments/base_modifications/">IGV desktop documentation</a> for more information.</li>
<li>basemod2 - Color alignments by base modification in two color mode. See the <a href="https://igv.org/doc/desktop/#UserGuide/tracks/alignments/base_modifications/">IGV desktop documentation</a> for more information.</li>
</ul>
<p><br/></p>
<h4 id="groupby-options">groupBy options<a class="headerlink" href="#groupby-options" title="Direct link to this section">#</a></h4>
<ul>
<li>none</li>
<li>strand - Group by read strand</li>
<li>firstOfPairStrand - For paired alignments.  Group both mates in a pair by the read strand of the mate marked "first in pair"</li>
<li>pairOrientation - For paired alignments.  Group by relative orientation (strand) of mates to each other</li>
<li>mateChr - For paired alignments.  Group alignments by the chromosome of its mate.</li>
<li>chimeric - Group by chimeric status.  An alignment is considered chimeric if it contains an 'SA' tag.</li>
<li>supplementary - Group by status of the SAM format "supplementary" tag</li>
<li>readOrder - For paired alignments, group by mate read order (first or second)</li>
<li>tag:<em>tagName</em> - Group alignments by the indicated tag.</li>
<li>base:<em>position</em> - Group alignments by the read base at the given position.  Position should be of the form <em>chromosomeName</em>:<em>basePosition</em>, e.g. base:chr1:1000 </li>
</ul>
<p><br/></p>
<h4 id="displaymode-options">displayMode options<a class="headerlink" href="#displaymode-options" title="Direct link to this section">#</a></h4>
<ul>
<li>EXPANDED - Pack alignments densely and draw at <em>alignmentRowHeight</em></li>
<li>SQUISHED - Pack alignments densely and draw at <em>squishedRowHeight</em></li>
<li>FULL - Draw 1 alignment per row at <em>alignmentRowHeight</em>.  </li>
</ul>
<p><br/></p>
<h4 id="paired-end-and-mate-pair-coloring-options">Paired-end and mate-pair coloring options.<a class="headerlink" href="#paired-end-and-mate-pair-coloring-options" title="Direct link to this section">#</a></h4>
<table>
<thead>
<tr>
<th>Property</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td>pairOrientation</td>
<td>Expected orientation of pairs, one of ff, fr, or rf.</td>
<td>Computed</td>
</tr>
<tr>
<td>minTLEN</td>
<td>Minimum expected absolute "TLEN" value.  Pairs below this value are colored blue if colorBy == "tlen" or "unexpectedPair"</td>
<td></td>
</tr>
<tr>
<td>maxTLEN</td>
<td>Maximum expected absolute "TLEN" value.   Pairs above this value are colored red if colorBy == "tlen" or "unexpectedPair"</td>
<td>Computed</td>
</tr>
<tr>
<td>minTLENPercentile</td>
<td>The percentile threshold for expected insert size.  If minTLEN is not specified this value is used to compute one. See notes below.</td>
<td>0.1</td>
</tr>
<tr>
<td>maxTLENPercentile</td>
<td>The percentile maximum for expected insert size.  If maxTLEN is not specified this value is used to compute one. See notes below.</td>
<td>99.9</td>
</tr>
</tbody>
</table>
<p><strong>Note on TLEN:</strong> This refers to column 9 of an alignment record from the SAM specification. It has variously been called "template length", "insert size", and "fragment length". There is no agreement on a precise definition, and aligner interpretations differ, but generally it can be thought of as the distance between start and end of an aligned pair.   Pairs with TLEN outside the expected range can be indicative of a deletion, or more rarely and insertion.</p>
<p><strong>Note on TLEN computation:</strong> if minTLEN or maxTLEN are not specified a value is computed from a sampling of the first data loaded for the track.  Specifically, the values are taken from a percentile of the sample data as specified by minPercentile and maxPercentile.  It is possible to specify an explicit value for one.</p>
<p><br/></p>
<h4 id="sort-option">Sort option<a class="headerlink" href="#sort-option" title="Direct link to this section">#</a></h4>
<p>The <code>sort</code> object defines initial sort order of packed alignment rows based on an alignment property at a specified position</p>
<table>
<thead>
<tr>
<th>Property</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td>chr</td>
<td>Sequence (chromosome) name</td>
<td></td>
</tr>
<tr>
<td>position</td>
<td>Genomic position (integer)</td>
<td></td>
</tr>
<tr>
<td>option</td>
<td>Parameter to sort by.  One of 'BASE', 'STRAND', 'INSERT_SIZE', 'MATE_CHR', 'MQ', 'TAG'</td>
<td></td>
</tr>
<tr>
<td>tag</td>
<td>Tag name to sort by.  Include only if option = 'TAG</td>
<td></td>
</tr>
<tr>
<td>direction</td>
<td>Sort directions.  ASC = ascending, DESC = descending</td>
<td>"ASC"</td>
</tr>
</tbody>
</table>
<p><br/></p>
<h4 id="filter-options">Filter options<a class="headerlink" href="#filter-options" title="Direct link to this section">#</a></h4>
<p>The <code>filter</code> object defines an alignment filter  based on sam flags.  If not supplied default filters are set 
as indicated in the table below.</p>
<table>
<thead>
<tr>
<th>Property</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td>vendorFailed</td>
<td>filter alignments marked as failing vendor quality checks (bit 0x200)</td>
<td>true</td>
</tr>
<tr>
<td>duplicates</td>
<td>filter alignments marked as a duplicate (bit 0x400)</td>
<td>true</td>
</tr>
<tr>
<td>secondary</td>
<td>filter alignments marked secondary (bit 0x100)</td>
<td>false</td>
</tr>
<tr>
<td>supplementary</td>
<td>filter alignments marked as supplmentary (bit 0x800)</td>
<td>false</td>
</tr>
<tr>
<td>mq</td>
<td>filter alignments with mapping quality &lt; supplied value (a number)</td>
<td>0</td>
</tr>
<tr>
<td>readgroups</td>
<td>array of read group names ('RG' tag).  If present filter alignments not matching this set</td>
<td></td>
</tr>
</tbody>
</table>
<p><br/></p>
<h2 id="functions">Functions<a class="headerlink" href="#functions" title="Direct link to this section">#</a></h2>
<p>An alignment track object exports 2 functions that can be called after track creation. </p>
<ul>
<li>setHighlightedReads(arrayOfReadNames, color)</li>
</ul>
<p>Highlights alignments with specified read names by drawing an outline in the specified color.</p>
<pre><code>bamTrack.setHighlightedReads([&quot;SRR099953.99059361&quot;, &quot;SRR099953.79101554&quot;], &quot;#0000ff&quot;)
</code></pre>
<ul>
<li>sort(sortOptions)</li>
</ul>
<p>Sort the alignment track.  See above for the <code>sortOptions</code> object definition.</p>
<pre><code>                bamTrack.sort({
                    chr: &quot;chr1&quot;,
                    position: 155155361,
                    option: &quot;BASE&quot;,
                    direction: &quot;ASC&quot;
                })
</code></pre>
<h2 id="example">Example<a class="headerlink" href="#example" title="Direct link to this section">#</a></h2>
<pre><code class="language-json">{
  &quot;type&quot;: &quot;alignment&quot;,
  &quot;format&quot;: &quot;bam&quot;,  
  &quot;name&quot;: &quot;NA12878&quot;,
  &quot;url&quot;: &quot;gs://genomics-public-data/platinum-genomes/bam/NA12878_S1.bam&quot;,
  &quot;indexURL&quot;: &quot;gs://genomics-public-data/platinum-genomes/bam/NA12878_S1.bam.bai&quot;,
}
</code></pre>

  <br>
</div>

<footer class="container-fluid wm-page-content">
  <!--
     <p>
       <a href="https://github.com/igvteam/igv.js-docs/edit/master/docs/tracks/Alignment-Track.md"><i class="fa fa-github"></i>
Edit on GitHub</a>
      </p>
  -->
  <!-- <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a> using <a href="https://github.com/gristlabs/mkdocs-windmill">Windmill</a> theme by Grist Labs.</p> -->
</footer>

</body>
</html>