<!DOCTYPE html>
<html lang="en">
<head>
  
  
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    
    
    
    <link rel="shortcut icon" href="../img/favicon.ico">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
    <title>Reference genome - igv.js</title>
    <link href="../css/bootstrap-3.3.7.min.css" rel="stylesheet">
    <link href="../css/font-awesome-4.7.0.css" rel="stylesheet">
    <link href="../css/base.css" rel="stylesheet">
    <link rel="stylesheet" href="../css/highlight.css">
    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
        <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
        <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <script src="../js/jquery-3.2.1.min.js"></script>
    <script src="../js/bootstrap-3.3.7.min.js"></script>
    <script src="../js/highlight.pack.js"></script>
    
    <base target="_top">
    <script>
      var base_url = '..';
      var is_top_frame = false;
        
        var pageToc = [
          {title: "genome property", url: "#_top", children: [
          ]},
          {title: "reference object", url: "#reference-object", children: [
              {title: "example", url: "#example" },
              {title: "details", url: "#details" },
          ]},
        ];

    </script>
    <script src="../js/base.js"></script> 
</head>

<body>
<script>
if (is_top_frame) { $('body').addClass('wm-top-page'); }
</script>



<div class="container-fluid wm-page-content">
  <a name="_top"></a>
    

    <p class="page-title">Genome</p>

<p>In igv.js a <em>genome</em> refers to a reference sequence and, optionally, one or more associated tracks and metadata such
as chromosome aliases and chromosome cytobands. </p>
<p>The reference genome can be defined with either the <strong>genome</strong> or <strong>reference</strong>  property.   One of these options is required.   </p>
<h2 id="genome-property"><em>genome</em> property<a class="headerlink" href="#genome-property" title="Direct link to this section">#</a></h2>
<p>The <em>genome</em> property specifies a reference genome by  an identifier for either (1) an IGV hosted genome definition,
or (2) a UCSC GenArk assembly.   </p>
<p>Example - IGV hosted genome</p>
<pre><code class="language-javascript">{
  ...
  genome: &quot;hg38&quot;,
  ...
}
</code></pre>
<p>Example - UCSC GenArk assembly</p>
<pre><code class="language-javascript">{
  ...
  genome: &quot;GCA_000002305.1&quot;
  ...
}

</code></pre>
<p>IGV hosts a limited number of genome assemblies, which are listed in the table below.   The UCSC GenArk site hosts
an extensive list of assembly hubs, currenlty numbering 4,247.  The complete listing can be 
found <a href="https://hgdownload.soe.ucsc.edu/hubs/UCSC_GI.assemblyHubList.txt">here</a>.</p>
<p><strong>IGV Hosted Genomes</strong></p>
<table>
<thead>
<tr>
<th>id</th>
<th>name</th>
</tr>
</thead>
<tbody>
<tr>
<td>hs1</td>
<td>Human (T2T CHM13-v2.0/hs1)</td>
</tr>
<tr>
<td>chm13v1.1</td>
<td>Human (T2T CHM13-v1.1)</td>
</tr>
<tr>
<td>hg38</td>
<td>Human (GRCh38/hg38)</td>
</tr>
<tr>
<td>hg38_1kg</td>
<td>Human (hg38 1kg/GATK)</td>
</tr>
<tr>
<td>hg19</td>
<td>Human (GRCh37/hg19)</td>
</tr>
<tr>
<td>hg18</td>
<td>Human (hg18)</td>
</tr>
<tr>
<td>mm39</td>
<td>Mouse (GRCm39/mm39)</td>
</tr>
<tr>
<td>mm10</td>
<td>Mouse (GRCm38/mm10)</td>
</tr>
<tr>
<td>mm9</td>
<td>Mouse (NCBI37/mm9)</td>
</tr>
<tr>
<td>rn7</td>
<td>Rat (rn7)</td>
</tr>
<tr>
<td>rn6</td>
<td>Rat (RGCS 6.0/rn6)</td>
</tr>
<tr>
<td>gorGor6</td>
<td>Gorilla (Kamilah_GGO_v0/gorGor6)</td>
</tr>
<tr>
<td>gorGor4</td>
<td>Gorilla (gorGor4.1/gorGor4)</td>
</tr>
<tr>
<td>panTro6</td>
<td>Chimp (panTro6) (panTro6)</td>
</tr>
<tr>
<td>panTro5</td>
<td>Chimp (panTro5) (panTro5)</td>
</tr>
<tr>
<td>panTro4</td>
<td>Chimp (SAC 2.1.4/panTro4)</td>
</tr>
<tr>
<td>macFas5</td>
<td>Macaca fascicularis (macFas5)</td>
</tr>
<tr>
<td>GCA_011100615.1</td>
<td>Macaca fascicularis 6.0 (GCA_011100615.1)</td>
</tr>
<tr>
<td>panPan2</td>
<td>Bonobo (MPI-EVA panpan1.1/panPan2)</td>
</tr>
<tr>
<td>canFam3</td>
<td>Dog (Broad CanFam3.1/canFam3)</td>
</tr>
<tr>
<td>canFam4</td>
<td>Dog (UU_Cfam_GSD_1.0/canFam4)</td>
</tr>
<tr>
<td>canFam5</td>
<td>Dog (canFam5)</td>
</tr>
<tr>
<td>bosTau9</td>
<td>Cow (ARS-UCD1.2/bosTau9)</td>
</tr>
<tr>
<td>bosTau8</td>
<td>Cow (UMD_3.1.1/bosTau8)</td>
</tr>
<tr>
<td>susScr11</td>
<td>Pig (SGSC Sscrofa11.1/susScr11)</td>
</tr>
<tr>
<td>galGal6</td>
<td>Chicken (galGal6)</td>
</tr>
<tr>
<td>GCF_016699485.2</td>
<td>Gallus gallus (GCF_016699485.2)</td>
</tr>
<tr>
<td>danRer11</td>
<td>Zebrafish (GRCZ11/danRer11)</td>
</tr>
<tr>
<td>danRer10</td>
<td>Zebrafish (GRCZ10/danRer10)</td>
</tr>
<tr>
<td>ce11</td>
<td>C. elegans (ce11)</td>
</tr>
<tr>
<td>dm6</td>
<td>D. melanogaster (dm6)</td>
</tr>
<tr>
<td>dm3</td>
<td>D. melanogaster (dm3)</td>
</tr>
<tr>
<td>dmel_r5.9</td>
<td>D. melanogaster (dmel_r5.9)</td>
</tr>
<tr>
<td>sacCer3</td>
<td>S. cerevisiae (sacCer3)</td>
</tr>
<tr>
<td>ASM294v2</td>
<td>S. pombe (ASM294v2)</td>
</tr>
<tr>
<td>ASM985889v3</td>
<td>Sars-CoV-2 (ASM985889v3)</td>
</tr>
<tr>
<td>tair10</td>
<td>A. thaliana (TAIR 10)</td>
</tr>
<tr>
<td>GCA_003086295.2</td>
<td>Peanut (GCA_003086295.2)</td>
</tr>
<tr>
<td>GCF_001433935.1</td>
<td>O. sativa IRGSP-1.0 (GCF_001433935.1)</td>
</tr>
<tr>
<td>NC_016856.1</td>
<td>Salmonella enterica subsp. enterica serovar Typhimurium str. 14028S</td>
</tr>
<tr>
<td>GCA_000182895.1</td>
<td>Coprinopsis cinerea okayama7#130 (GCA_000182895.1)</td>
</tr>
</tbody>
</table>
<h2 id="reference-object"><em>reference</em> object<a class="headerlink" href="#reference-object" title="Direct link to this section">#</a></h2>
<p>To define or customize a reference genome the reference property can be used.  </p>
<h3 id="example">example<a class="headerlink" href="#example" title="Direct link to this section">#</a></h3>
<pre><code class="language-javascript">
{
  reference: {
    &quot;id&quot;: &quot;hg38&quot;,
    &quot;name&quot;: &quot;Human (GRCh38/hg38)&quot;,
    &quot;fastaURL&quot;: &quot;https://s3.amazonaws.com/igv.broadinstitute.org/genomes/seq/hg38/hg38.fa&quot;,
    &quot;indexURL&quot;: &quot;https://s3.amazonaws.com/igv.broadinstitute.org/genomes/seq/hg38/hg38.fa.fai&quot;,
    &quot;cytobandURL&quot;: &quot;https://s3.amazonaws.com/igv.broadinstitute.org/annotations/hg38/cytoBandIdeo.txt&quot;,
    &quot;tracks&quot;: [
      {
        &quot;name&quot;: &quot;Refseq Genes&quot;,
        &quot;url&quot;: &quot;https://s3.amazonaws.com/igv.org.genomes/hg38/refGene.txt.gz&quot;,
        &quot;order&quot;: 1000000,
        &quot;indexed&quot;: false
      }
    ]
  }
}
</code></pre>
<h3 id="details">details<a class="headerlink" href="#details" title="Direct link to this section">#</a></h3>
<p>All properties except a reference sequence URL (fastaURL or twoBitURL) are optional.   An index is highly recommended 
for fasta files if the sequence is &gt; a few mb in size.   Loading a large fasta file without an index is likely to 
freeze the browser.  In total there are 4 options for defining the reference sequence:</p>
<ol>
<li>Fasta file without index, useful for small (&lt; 10 MB) sequences.  <strong>fastaURL</strong></li>
<li>Indexed fasta file.  <strong>fastaURL, indexURL</strong></li>
<li>BGZIPPED compressed fasta file.  <strong>fastaURL, indexURL, compressedIndexURL</strong></li>
<li>UCSC two bit sequence file (see [twobit]). (https://genome.ucsc.edu/goldenPath/help/twoBit.html).  <strong>twoBitURL</strong></li>
</ol>
<table>
<thead>
<tr>
<th>Option</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td>id</td>
<td>UCSC or other id string.</td>
<td></td>
</tr>
<tr>
<td>name</td>
<td>A descriptive name.</td>
<td></td>
</tr>
<tr>
<td>fastaURL</td>
<td>URL to a FASTA file. <strong>Either fastaURL or twoBitURL is required</strong>.</td>
<td></td>
</tr>
<tr>
<td>indexURL</td>
<td>URL to a FASTA index (.fai file).  An index file is optional, but if not supplied the entire fasta is read.</td>
<td></td>
</tr>
<tr>
<td>compressedIndexURL</td>
<td>URL to a BGZIPPED compressed fasta file (.gzi index).  Note this is in addition, not in place, of the .fai index.</td>
<td></td>
</tr>
<tr>
<td>twoBitURL</td>
<td>URL to a UCSC two-bit sequence file (Version 3.0).  Required if fastaURL is not defined.  If both fastaURL and twoBitURL are defined twoBitURL will be used for igv.js version &gt;= 3.0, fastaURL for earlier igv.js versions.</td>
<td></td>
</tr>
<tr>
<td>cytobandURL</td>
<td>URL to a cytoband ideogram file in UCSC format.  <em>Optional</em></td>
<td></td>
</tr>
<tr>
<td>aliasURL</td>
<td>URL to a tab-delimited file defining aliases for chromosome names.  Theformat is tab delimited with  1 line per chromosome.  Each line begins with the canonical chromosome followed by aliases.  See the example below.   <em>Optional</em></td>
<td></td>
</tr>
<tr>
<td>chromSizesURL</td>
<td>URL to a UCSC .chrom.sizes file.   This can be used in combination with twoBitURL to support the whole genome view.</td>
<td></td>
</tr>
<tr>
<td>indexed</td>
<td>Flag indicating if the FASTA is indexed.  Ignored if indexURL is supplied.  The primary purpose of this property is to indicate that the fasta is not indexed.  <em><strong>Deprecated as of Version 3.0</strong></em></td>
<td></td>
</tr>
<tr>
<td>tracks</td>
<td>A list of tracks to be loaded with the genome.  See the <a href="../tracks/Tracks/">tracks</a> description for details.  <em>Optional</em></td>
<td></td>
</tr>
<tr>
<td>chromosomeOrder</td>
<td>An array of chromosome names defining the order in the whole genome view and chromosome pulldown selector, if used.   <em>Optional</em></td>
<td></td>
</tr>
<tr>
<td>headers</td>
<td>HTTP headers to include with each request. For example {"authorization": "bearer: token"}.  <em>Optional</em></td>
<td></td>
</tr>
<tr>
<td>wholeGenomeView</td>
<td>Construct a "whole genome" view from the individual sequences.  This is useful for finished assemblies with a few (&lt; 50) large chromosomes.  Its not useful for assemblies with a single or conversely thousands of sequences. <em>Optional</em></td>
<td>true</td>
</tr>
</tbody>
</table>
<h4 id="chromosome-alias-example">Chromosome alias example<a class="headerlink" href="#chromosome-alias-example" title="Direct link to this section">#</a></h4>
<p>Below is the contents of a chromosome alias file.</p>
<pre><code>chr1    1   CM000663.2  NC_000001.11
chr2    2   CM000664.2  NC_000002.12
chr3    3   CM000665.2  NC_000003.12
chr4    4   CM000666.2  NC_000004.12
chr5    5   CM000667.2  NC_000005.10
chr6    6   CM000668.2  NC_000006.12
chr7    7   CM000669.2  NC_000007.14
chr8    8   CM000670.2  NC_000008.11
chr9    9   CM000671.2  NC_000009.12
chr10   10  CM000672.2  NC_000010.11
chr11   11  CM000673.2  NC_000011.10
chr12   12  CM000674.2  NC_000012.12
chr13   13  CM000675.2  NC_000013.11
chr14   14  CM000676.2  NC_000014.9
chr15   15  CM000677.2  NC_000015.10
chr16   16  CM000678.2  NC_000016.10
chr17   17  CM000679.2  NC_000017.11
chr18   18  CM000680.2  NC_000018.10
chr19   19  CM000681.2  NC_000019.10
chr20   20  CM000682.2  NC_000020.11
chr21   21  CM000683.2  NC_000021.9
chr22   22  CM000684.2  NC_000022.11
chrX    X   CM000685.2  NC_000023.11    23
chrY    Y   CM000686.2  NC_000024.10    24
chrM    MT  J01415.2    NC_012920.1 chrMT
</code></pre>

  <br>
</div>

<footer class="container-fluid wm-page-content">
  <!--
     <p>
       <a href="https://github.com/igvteam/igv.js-docs/edit/master/docs/Reference-Genome.md"><i class="fa fa-github"></i>
Edit on GitHub</a>
      </p>
  -->
  <!-- <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a> using <a href="https://github.com/gristlabs/mkdocs-windmill">Windmill</a> theme by Grist Labs.</p> -->
</footer>

</body>
</html>